### PR TITLE
TEST(day_12): Add arrange_spring ut

### PR DESCRIPTION
**Pull request by dariannax3**
Function arrange_spring should return proper spring arragement when extracted conditions and groups are provided.
Function will return proper value for data when the arragements is equal 1.

Two helper functions are expected to be add, to extrat spring conditions and spring groups from provided data.

### CODE REVIEW

File: day_12/task_1/hot_springs.cpp [hunk 1]
```diff
@@ -1 +1,47 @@
...
+#include <sstream>
+#include <algorithm>
+
+std::vector<int> extract_groups(std::string spring_condition_data)
+{
+    std::replace(spring_condition_data.begin(), spring_condition_data.end(),',', ' ');
+    std::stringstream ss{spring_condition_data};
+    std::string conditions;
+    std::vector<int> groups(3);
+    ss >> conditions >> groups[0] >> groups[1] >> groups[2];
```
Comment (dariannax3) [inline day_12/task_1/hot_springs.cpp]: what happened when will be 4 groups?

File: day_12/task_1/hot_springs.cpp [hunk 2]
```diff
@@ -1 +1,47 @@
...
+    {
+        while(extracted_condition[i] == '?')
+        {
+            if(extracted_spring_groups[current_group] > 0)
+            {
+                extracted_condition[i] = '#';
+                --(extracted_spring_groups[current_group]);
+                ++i;
+
+                if(extracted_spring_groups[current_group] == 0 && extracted_condition[i] == '?')
```
Comment (dariannax3) [inline day_12/task_1/hot_springs.cpp:44]: could you explain why extracted_spring_groups[current_group] == 0?

File: day_12/task_1/hot_springs.cpp (hunk 1 above)
Comment (Rosein) [reply day_12/task_1/hot_springs.cpp]: Then it would be hard to modify. So I changed it into for loop.

File: day_12/task_1/hot_springs.cpp (hunk 2 above)
Comment (Rosein) [reply day_12/task_1/hot_springs.cpp:44]: As you can see in the line 34, we decrease iteratively `extracted_spring_groups[current_group] value`. 
`extracted_spring_groups[current_group] == 0` is to check ending condition, whether we reach last position from specific group.

File: day_12/task_1/hot_springs.cpp [hunk 3]
```diff
@@ -1 +1,54 @@
...
+
+std::vector<int> extract_groups(std::string spring_condition_data)
+{
+    std::replace(spring_condition_data.begin(), spring_condition_data.end(),',', ' ');
+    std::stringstream ss{spring_condition_data};
+    std::string conditions;
+    ss >> conditions;
+
+    static const auto kSizeOfGroups = 3U;
+    std::vector<int> groups(kSizeOfGroups);
```
Comment (dariannax3) [inline day_12/task_1/hot_springs.cpp]: make it more general

**Pull request merged**